torch/_functorch/autograd_function.py

```diff
@@ -454,18 +453,31 @@ def jvp_no_context(saved_tensors, tangents):
         tangent_in_dims = get_tangents_in_dims(in_dims, tangents)
         out_tangents, out_tangents_dims = restore_vmap(
             jvp_no_context,
-            (saved_tensors_bdims, tangent_in_dims),
+            (ctx.saved_tensors_bdims[key], tangent_in_dims),
             batch_size,
             randomness,
         )(ctx.saved_tensors, tangents)
 
-        result = reductify(out_tangents, out_tangents_dims, out_dims, batch_size)
-        return result
+        result = reductify(
+            out_tangents, out_tangents_dims, ctx.out_dims[key], batch_size
+        )
+        if isinstance(result, torch.Tensor):
+            return result, None
+        else:
+            return *result, None
 
     def backward(ctx, *grad_outputs):
-        assert out_dims != init_val
-        assert input_shapes != init_val
-        assert saved_tensors_bdims != init_val
+        key = id(Generated)
```

Comment:
I'm really confused about if this is actually capturing the correct Generated object. Like, there is no `Generated` object in scope until it actually gets created later, are these nested functions capturing the correct thing?

Reply:
nvm I see